llvm/lib/CheerpWriter/CheerpWasmWriter.cpp

```diff
@@ -1705,6 +2092,153 @@ bool CheerpWasmWriter::compileInlineInstruction(WasmBuffer& code, const Instruct
 						encodeInst(WasmS32Opcode::I32_CONST, id, code);
 						return false;
 					}
+					case Intrinsic::wasm_shuffle:
+					{
+						compileOperand(code, ci.getOperand(0));
+						compileOperand(code, ci.getOperand(1));
+						encodeInst(WasmSIMDOpcode::I8x16_SHUFFLE, code);
+						assert(ci.getNumOperands() > 17);
+						for (int i = 2; i < 18; i++)
+						{
+							const Value* v = ci.getOperand(i);
+							assert(isa<ConstantInt>(v));
+							const ConstantInt *num = dyn_cast<ConstantInt>(v);
+							code << static_cast<char>(num->getZExtValue());
+						}
+						return false;
+					}
+					case Intrinsic::wasm_swizzle:
+					{
+						compileOperand(code, ci.getOperand(0));
+						compileOperand(code, ci.getOperand(1));
+						encodeInst(WasmSIMDOpcode::I8x16_SWIZZLE, code);
+						return false;
+					}
+					case Intrinsic::wasm_anytrue:
+					{
+						compileOperand(code, ci.getOperand(0));
+						encodeInst(WasmSIMDOpcode::V128_ANY_TRUE, code);
+						return false;
+					}
+					case Intrinsic::wasm_bitselect:
+					{
+						compileOperand(code, ci.getOperand(0));
+						compileOperand(code, ci.getOperand(1));
+						compileOperand(code, ci.getOperand(2));
+						encodeInst(WasmSIMDOpcode::V128_BITSELECT, code);
+						return false;
+					}
+					case Intrinsic::minimum:
+					{
+						assert(ci.getOperand(0)->getType()->isVectorTy());
+						Type* et = cast<VectorType>(ci.getOperand(0)->getType())->getElementType();
+						assert(et->isFloatTy() || et->isDoubleTy());
+						compileOperand(code, ci.getOperand(0));
+						compileOperand(code, ci.getOperand(1));
+						if (et->isFloatTy())
+							encodeInst(WasmSIMDOpcode::F32x4_MIN, code);
+						else
+							encodeInst(WasmSIMDOpcode::F64x2_MIN, code);
+						return false;
+					}
+					case Intrinsic::maximum:
+					{
+						assert(ci.getOperand(0)->getType()->isVectorTy());
+						Type *et = cast<VectorType>(ci.getOperand(0)->getType())->getElementType();
+						assert(et->isFloatTy() || et->isDoubleTy());
+						compileOperand(code, ci.getOperand(0));
+						compileOperand(code, ci.getOperand(1));
+						if (et->isFloatTy())
+							encodeInst(WasmSIMDOpcode::F32x4_MAX, code);
+						else
+							encodeInst(WasmSIMDOpcode::F64x2_MAX, code);
+						return false;
```

Comment:
In all of those I believe it should taken care the case of isTailCall.
Something like:
						if(useTailCall)
							encodeInst(WasmOpcode::RETURN, code);
						return true;

Reply:
Done